elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala

```diff
@@ -86,6 +86,27 @@ class RestlasticSearchClientTest extends WordSpec with Matchers with ScalaFuture
       whenReady(mappingFut) { _ => refresh() }
     }
 
+    "Be able to setup document mapping with field index options" in {
+      val basicFieldDocsMapping = BasicFieldMapping(StringType, None, Some(analyzerName), None,
+        Some(analyzerName), fieldIndexOption = Some(DocsIndexOption))
+      val basicFieldFreqsMapping = BasicFieldMapping(StringType, None, Some(analyzerName), None,
+        Some(analyzerName), fieldIndexOption = Some(FreqsIndexOption))
+      val basicFieldPositionsMapping = BasicFieldMapping(StringType, None, Some(analyzerName), None,
+        Some(analyzerName), fieldIndexOption = Some(PositionsIndexOption))
+      val basicFieldOffsetsMapping = BasicFieldMapping(StringType, None, Some(analyzerName), None,
+        Some(analyzerName), fieldIndexOption = Some(OffsetsIndexOption))
+
+      val timestampMapping = EnabledFieldMapping(true)
+      val metadataMapping = Mapping(tpe, IndexMapping(
+        Map("name" -> basicFieldDocsMapping, "f1" -> basicFieldFreqsMapping,
+          "f2" -> basicFieldOffsetsMapping, "text" -> basicFieldOffsetsMapping,
+          "suggest" -> CompletionMapping(Map("f" -> CompletionContext("name")),
+            analyzerName)), timestampMapping))
+
+      val mappingFut = restClient.putMapping(index, tpe, metadataMapping)
+      whenReady(mappingFut) { _ => refresh() }
```

Comment:
This test only checks that you put the mapping and it happens to not error (if we even propagate errors properly!) Can you make a call to `getMapping` to verify that the mapping was actually applied?